### DISCUSSION TITLE
Expectation on Snowflake starts with schema scan

### DISCUSSION THREAD

**Discussion by razido1** [great-expectations/great_expectations#5343] [Support]
When running great expectation on Snowflake with SqlAlchemyExecutionEngine, before the validations starts, there is a scan of all the schemas in the given DB:
It starts with 
`SHOW /* sqlalchemy:get_schema_names */ SCHEMAS`
and then for each schema: 
```
SHOW /* sqlalchemy:get_table_names */ TABLES IN <schema>
SHOW /* sqlalchemy:get_view_names */ VIEWS IN <schema>
```
Is there a way to avoid this? our snowflake schemas are dynamic and can be deleted between the time it reads them and the time it checks for their tables/views. In addition, it is quite costly and not needed since our expectations are on a specific schema

**Comment by leszekgruchala** [comment]
It's been answered on a duplicate of this question https://github.com/great-expectations/great_expectations/discussions/5370
